.github/workflows/ompi_mpi4py.yaml

```
@@ -84,11 +91,58 @@ jobs:
       with:
         repository: "mpi4py/mpi4py"
 
-    - name: Install mpi4py
-      run:  python -m pip install .
+    - name: Build mpi4py wheel
+      run:  python -m pip wheel .
       env:
         CFLAGS: "-O0"
 
+    - name: Save the artifacts for other jobs
+      uses: actions/upload-artifact@v4
+      with:
+        # I dislike hard-coding /home/runner, but I couldn't get $HOME
+        # or ${{ env.HOME }} to expand properly.
+        #
+        # The "test" and "demo" directories are from mpi4py -- they
+        # aren't installed by the wheel.
+        path: |
+          /opt/openmpi
+          /home/runner/.openmpi
+          /home/runner/.prte
```

No, I think GITHUB_WORKSPACE is not the right path, you truly want `$HOME`, which is the place where the `.openmpi` and `.prte` config directories are located.

Can you try doing it this way as per the [documentation](https://github.com/actions/upload-artifact/?tab=readme-ov-file#environment-variables-and-tilde-expansion) :

```suggestion
          ~/.openmpi
          ~/.prte
```

---

That seemed to do the trick; thanks @dalcinl